src/main/java/org/zalando/putittorest/HttpClientFactoryBean.java

```diff
@@ -36,9 +40,17 @@ public void setSocketTimeout(final int socketTimeout) {
         config.setSocketTimeout(socketTimeout);
     }
 
+    @Hack("Hide type to avoid runtime dependency")
+    public void setMetricsWrapper(final Object metricsWrapper) {
+        this.metricsWrapper = metricsWrapper;
+    }
+
     @Override
     public CloseableHttpClient getObject() {
         builder.setConnectionTimeToLive(30, TimeUnit.SECONDS);
+        if (metricsWrapper != null) {
+            builder.setRequestExecutor(new InstrumentedHttpRequestExecutor((MetricsWrapper) metricsWrapper));
```

Comment:
I took a look and I think we should put the interceptors into `put-it-to-rest`. The project is already `httpcomponents` specific and has a dependency to `httpcore`.